tests/e2e/hci.go

```diff
@@ -488,4 +495,336 @@ var _ bool = ginkgo.Describe("hci", func() {
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Create 2 sts with 3 replica and 1 replica respectively")
+		sts := createCustomisedStatefulSets(ctx, client, namespace, false, replicas,
+			false, nil, false, true, "", "", sc, "")
+		defer func() {
+			ginkgo.By(fmt.Sprintf("Deleting all statefulsets in namespace: %v", namespace))
+			fss.DeleteAllStatefulSets(ctx, client, namespace)
+		}()
+
+		pods := fss.GetPodList(ctx, client, sts)
+		targetWorkerName := pods.Items[0].Spec.NodeName
+
+		ginkgo.By(fmt.Sprintf("Power off the node: %v", targetWorkerName))
+		vmUUID := getNodeUUID(ctx, client, targetWorkerName)
+		gomega.Expect(vmUUID).NotTo(gomega.BeEmpty())
+		framework.Logf("VM uuid is: %s for node: %s", vmUUID, targetWorkerName)
+		vmRef, err := e2eVSphere.getVMByUUID(ctx, vmUUID)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("vmRef: %+v for the VM uuid: %s", vmRef, vmUUID)
+		gomega.Expect(vmRef).NotTo(gomega.BeNil(), "vmRef should not be nil")
+		targetVm = object.NewVirtualMachine(e2eVSphere.Client.Client, vmRef.Reference())
+		_, err = targetVm.PowerOff(ctx)
+		framework.ExpectNoError(err)
+
+		err = targetVm.WaitForPowerState(ctx, vimtypes.VirtualMachinePowerStatePoweredOff)
+		framework.ExpectNoError(err, "Unable to power off the node")
+		isK8sNodePoweredOff = true
+
+		defer func() {
+			if isK8sNodePoweredOff {
+				ginkgo.By("Re-integrate isolated host back into vsan cluster before terminating test")
+				_, err = targetVm.PowerOn(ctx)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				isK8sNodePoweredOff = false
+			}
+		}()
+		taint := &v1.Taint{
+			Key:    "node.kubernetes.io/out-of-service",
+			Value:  "nodeshutdown",
+			Effect: v1.TaintEffect("NoExecute"),
+		}
+		node, err := client.CoreV1().Nodes().Get(ctx, targetWorkerName, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		taintedNode, isNodeTainted, err := taints.AddOrUpdateTaint(node, taint)
+		gomega.Expect(isNodeTainted).To(gomega.BeTrue())
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		_, err = client.CoreV1().Nodes().Update(ctx, taintedNode, metav1.UpdateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			framework.Logf("Removing taint from node: %s", targetWorkerName)
+			node, err = client.CoreV1().Nodes().Get(ctx, targetWorkerName, metav1.GetOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			untaintedNode, isNodeUntainted, err := taints.RemoveTaint(node, taint)
+			gomega.Expect(isNodeUntainted).To(gomega.BeTrue())
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			_, err = client.CoreV1().Nodes().Update(ctx, untaintedNode, metav1.UpdateOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Scale up statefulset and scale down statefulset1 to 2 replicas")
+		scaleUpStsAndVerifyPodMetadata(ctx, client, namespace, sts, replicas+2, true, true)
```

Comment:
if the replica initialization is set to 1, the scale up replicas to 3.
Description needs to be corrected.

Reply:
Updated